src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -37,18 +42,37 @@ internal DirectoryInfo Root
             }
         }
 
-        internal TestAssetInfo(DirectoryInfo root, string assetName)
+        internal TestAssetInfo(DirectoryInfo root, string assetName, FileInfo dotnetExeFile, string projectFilePattern)
         {
             if (!root.Exists)
```

Comment:
The approach of validating the file system state in the constructor is inconsistent with the behavior of, e.g., `FileInfo` and `DirectoryInfo` and may be slightly counterintuitive for that reason. Those classes are in line with with the guidance of not doing work in the constructor. This class represents "info" about a "test asset". The existence of the corresponding files is, to my mind, just info, and not a prerequisite for the instantiation of the class.